plugins/plugin-maven/che-plugin-maven-server/src/main/java/org/eclipse/che/plugin/maven/server/inject/MavenModule.java

```diff
@@ -39,7 +39,10 @@
     protected void configure() {
         newSetBinder(binder(), ValueProviderFactory.class).addBinding().to(MavenValueProviderFactory.class);
 
-        newSetBinder(binder(), ProjectTypeDef.class).addBinding().to(MavenProjectType.class);
+        //bind maven project type only if maven installed on dev machine
+        if(System.getenv("M2_HOME") != null) {
```

Comment:
AFAIK usually we inject named variable for getting environment variable's value
`@Named("env.M2_HOME") String m2Home`